community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/Comparer.scala

```diff
@@ -40,6 +40,18 @@ trait Comparer extends CypherSerializer {
         throw new IncomparableValuesException(serializeWithType(l), serializeWithType(r))
```

Comment:
If `l` or `r` is `null` this will throw null pointer exception (from serializeWithType) instead of the intended exception.